src/audio/lib/thread.c

```diff
@@ -790,13 +790,13 @@ void AudioThread_ProcessChannelCmd(SequenceChannel* channel, AudioCmd* cmd) {
             break;
 
         case AUDIOCMD_OP_CHANNEL_SET_SFX_STATE:
-            channel->sfxState = cmd->asUInt;
+            channel->sfxState = cmd->data;
```

Comment:
I think `cmd->data` is suppose to be the internal variable not accessed by the commands. The type should be paired with whatever function it's tied to:
`AudioThread_QueueCmdF32`
`AudioThread_QueueCmdS32`
`AudioThread_QueueCmdS8`
`AudioThread_QueueCmdU16`
(although S32 covers both s32/u32, U16 covers both s16/u16, and S8 covers both s8/u8, names should reflect that).

What would the casting look like without using `data` for everything?

Reply:
It looks like this https://github.com/zeldaret/mm/pull/1266/commits/af9e263c96d9fdfab772c8327bdb2638ef9b7292
I'm not a huge fan of stuff like this
```c
gAudioCustomUpdateFunction = (AudioCustomUpdateFunction)cmd->asUInt;
```
casting a number to a function seems fishy

Reply:
Yeah, that is kinda ugly. I wonder if it's worth adding one more member `cmd->asPtr` and define `asPtr` as `void*`?

Reply:
yeah, that sounds like a better solution, I'll go with it